app/docs/0.12.x/healthchecks.md

```diff
@@ -0,0 +1,182 @@
+---
+title: Health Checks Reference
+---
+
+# Health Checks Reference
+
+You can make an API proxied by Kong use a [ring-balancer][1], configuring
+by adding an [upstream][2] entity that contains one or more [target][3]
+entities, each target pointing to a different IP address (or hostname) and
+port. The ring-balancer will balance load among the various targets, and based
+on the [upstream][2] configuration, will perform health checks on the targets,
+making them as healthy or unhealthy whether they are responsive or not. The
+ring-balancer will then only route traffic to healthy targets.
+
+Kong supports two kinds of health checks, which can be used separately or in
+conjunction:
+
+* **active checks**, where a specific endpoint in the target is periodically
+requested and the health of the target is determined based on its response;
+
+* **passive checks** (also known as **circuit breakers**), where Kong analyzes
+the ongoing traffic being proxied and determines the health of targets based
+on their behavior responding requests.
+
+[1]: /docs/{{page.kong_version}}/loadbalancing#ring-balancer
+[2]: /docs/{{page.kong_version}}/loadbalancing#upstream
+[3]: /docs/{{page.kong_version}}/loadbalancing#ring-balancer
+
+### Table of Contents
+
+- [Healthy and unhealthy targets](#healthy-and-unhealthy-targets)
+- [Active health checks](#active-health-checks)
+- [Passive health checks (circuit breakers)](#passive-health-checks-circuit-breakers)
+- [Summary of pros and cons](#summary-of-pros-and-cons)
+
+### Healthy and unhealthy targets
+
+The objective of the health checks functionality is to dynamically mark
+targets as **healthy** or **unhealthy**, **for a given Kong node**. There is
+no cluster-wide synchronization of health information: each Kong node
+determines the health of its targets separately. This is desirable since at a
+given point one Kong node may be able to connect to a target successfully
+while another node is failing to reach it: the first node will consider
+it healthy, while the second will mark it as unhealthy and start routing
+traffic to other targets of the upstream.
+
+Either an active probe (on active health checks) or a proxied request
+(on passive health checks) produces data which is used to determine
+whether a target is healthy or unhealthy. A request may produce a TCP
+error, timeout, or produce an HTTP status code. Based on this
+information, the health checker updates a series of internal counters:
+
+* If the returned status code is one configured as "healthy", it will
+increment the "Successes" counter for the target and clear all its other
```

Comment:
would replace quote-wrapping of "Successes" with monospace formatting since it represents a programmatic structure element (and idem throughout this whole section)

Reply:
errm, I guess I wrote it that way because to me it's more conceptual than programmatic, since there's nothing called `Successes` (in this exact phrasing) anywhere. I didn't want to confuse it with the `successes` field which is not the counter, but the counter threshold value.

Reply:
My mistake! Stylistically I would still not wrap in quotes, but that's just how my brain reads it. Not a blocker for me.